org.eclipse.xtend.core/src/org/eclipse/xtend/core/validation/XtendValidator.java

```diff
@@ -500,10 +500,11 @@ protected EObject getContainingAnnotationTarget(XAnnotation annotation) {
 	}
 
 	@Check
-	public void checkNoVoidInDependencyDeclaration(XtendField dep) {
-		JvmTypeReference declaredFieldType = dep.getType();
+	public void checkXtendFieldNotPrimitiveVoid(XtendField field) {
+		JvmTypeReference declaredFieldType = field.getType();
 		if (isPrimitiveVoid(declaredFieldType)) {
-			error("Primitive void cannot be a dependency.", dep.getType(), null, INVALID_USE_OF_TYPE);
+			error("void is an invalid type for the field " + field.getName(),
```

Comment:
I see that the message was take from JDT. In Xbase, we use 
```
@Check
	public void checkTypeReferenceIsNotVoid(XExpression expression) {
		for (EObject eObject : expression.eContents()) {
			if (eObject instanceof JvmTypeReference) {
				JvmTypeReference typeRef = (JvmTypeReference) eObject;
				if (isPrimitiveVoid(typeRef)) {
					if (typeRef.eClass() == TypesPackage.Literals.JVM_PARAMETERIZED_TYPE_REFERENCE) {
						if (!((JvmParameterizedTypeReference) typeRef).getArguments().isEmpty()) {
							continue;
						}
					}
					error("Primitive void cannot be used here.", typeRef, null, INVALID_USE_OF_TYPE);
				}
			}
		}
	}
```

and in common.types

```
	protected void checkNotPrimitive(JvmTypeReference jvmTypeReference) {
		if (primitives.isPrimitive(jvmTypeReference)) {
			error("The primitive '"+jvmTypeReference.getQualifiedName('.')+"' cannot be a type argument", jvmTypeReference, null, IssueCodes.INVALID_USE_OF_TYPE);
		}
	}
```

All not super aligned with each other. Maybe keep it somewhat consistent and use `Primitive void is an invalid type for the field ...`

Reply:
@szarnekow OK, I'll change it to `Primitive void is an invalid type for the field` and then merge it, OK?